File: SearchInsertPosition.java
/*
Runtime: 0 ms, faster than 100.00% of Java online submissions for Search Insert Position.
Memory Usage: 41.8 MB, less than 90.75% of Java online submissions for Search Insert Position.
*/
class Solution {
    public int searchInsert(int[] nums, int target) {
        
        int start = 0;
        int end = nums.length - 1;
        
        while(start <= end){
            
            int mid = start + (end - start) / 2;
            
            if(nums[mid] >= target){
                end = mid - 1;
            }else{
                start = mid + 1;
            }
            
        }
        
        return start;
        
    }
}

File: SearchInsertPosition.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let start = 0, end = nums.length;
    while(start < end) {
        const mid = start + Math.floor((end - start) / 2);
        if(nums[mid] === target) return mid;
        if(nums[mid] < target) start = mid + 1;
        else end = mid;
    }
    return start;
};
